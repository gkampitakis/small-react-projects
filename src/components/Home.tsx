import React, { ReactElement } from 'react';
import routes from '../routes';
import { Link } from 'react-router-dom';


export default function Home (): ReactElement {
  return (
    <main>
      <header >
        Small React Projects
      </header>
      <section>
        {routes.map(({ projectName, route }, idx) => (
          <Link key={idx} to={route}>
            {projectName}
          </Link>
        ))}
      </section>
      <footer>
        G Kampitakis
      </footer>
    </main>
  );
}
