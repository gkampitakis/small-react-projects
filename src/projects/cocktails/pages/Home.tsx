import React, { ReactElement } from 'react';
import { useGlobalContext } from '../context';
import Search from '../components/Search';
import List from '../components/List';
import ErrorCmp from '../../../components/Error';
import Loading from '../../../components/Loading';

export default function Home(): ReactElement {
  const { loading, cocktailList, error } = useGlobalContext();
  const show = loading && !cocktailList.length;

  return (
    <>
      <Search />
      {!show && !error && <List items={cocktailList} />}
      {show && <Loading />}
      {error && <ErrorCmp message={error} />}
    </>
  );
}
