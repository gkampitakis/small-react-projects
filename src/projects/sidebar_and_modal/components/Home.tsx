import React, { ReactElement } from 'react';
import { FaBars } from 'react-icons/fa';
import { useGlobalContext } from '../context';


export default function Home (): ReactElement {
  const { openSidebar, openModal } = useGlobalContext();

  return (
    <main>
      <button onClick={openSidebar} className="sidebar-toggle" data-testid='open-sidebar'>
        <FaBars />
      </button>
      <button onClick={openModal} className="show-btn">
        show modal
      </button>
    </main>
  );
}
