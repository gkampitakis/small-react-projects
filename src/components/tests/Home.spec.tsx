import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Home from '../Home';


describe('Home', () => {
  it('Should render correctly', () => {
    const { baseElement } = render(<Home />, { wrapper: MemoryRouter });

    expect(baseElement).toMatchSnapshot();
  });
});
