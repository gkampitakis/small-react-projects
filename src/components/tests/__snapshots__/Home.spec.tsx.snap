// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Should render correctly 1`] = `
<body
  style="background: rgb(241, 241, 241); color: rgb(16, 16, 16);"
>
  <div>
    <main
      class="homePage"
    >
      <header>
        <h2>
          Small React Projects
        </h2>
      </header>
      <section>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Accordion Project"
            />
          </picture>
          <a
            href="/accordion"
          >
            Accordion
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Birthday Project"
            />
          </picture>
          <a
            href="/birthday"
          >
            Birthday
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Cart Project"
            />
          </picture>
          <a
            href="/cart"
          >
            Cart
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Cocktails Project"
            />
          </picture>
          <a
            href="/cocktails"
          >
            Cocktails
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Dark Mode Project"
            />
          </picture>
          <a
            href="/dark-mode"
          >
            Dark Mode
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Grocery Bud Project"
            />
          </picture>
          <a
            href="/grocery-bud"
          >
            Grocery Bud
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Lorem Generator Project"
            />
          </picture>
          <a
            href="/lorem-ipsum"
          >
            Lorem Generator
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Markdown Preview Project"
            />
          </picture>
          <a
            href="/markdown-preview"
          >
            Markdown Preview
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Menu Project"
            />
          </picture>
          <a
            href="/menu"
          >
            Menu
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Navbar Project"
            />
          </picture>
          <a
            href="/navbar"
          >
            Navbar
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Photo Album Project"
            />
          </picture>
          <a
            href="/photo-album"
          >
            Photo Album
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Quiz Project"
            />
          </picture>
          <a
            href="/quiz"
          >
            Quiz
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Random User Project"
            />
          </picture>
          <a
            href="/random-user"
          >
            Random User
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Reviews Project"
            />
          </picture>
          <a
            href="/reviews"
          >
            Reviews
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Sidebar and Modal Project"
            />
          </picture>
          <a
            href="/sidebar-modal"
          >
            Sidebar and Modal
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Slider Project"
            />
          </picture>
          <a
            href="/slider"
          >
            Slider
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Tabs Project"
            />
          </picture>
          <a
            href="/tabs"
          >
            Tabs
          </a>
        </div>
        <div
          class="project"
        >
          <picture>
            <source />
            <img
              alt="Tours Project"
            />
          </picture>
          <a
            href="/tours"
          >
            Tours
          </a>
        </div>
      </section>
      <footer>
        <span>
          Author:
        </span>
          gkampitakis
      </footer>
    </main>
  </div>
</body>
`;
